### PR TITLE
Fix multiple MantidPlot instances check

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

Fixes several issues:
* Catches an exception due to zombie processes seen on the OSX package
* Uses the executable filename rather than the application name to match against other processes

There is a caveat that development builds on Linux will not "see" proper package builds as the executable is renamed during package installation.

**To test:**

Test using a package on all 3 operating systems.

*No associated issue*

Does this update require release notes?
- [ ] Yes
- [X] No

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by NickDraper — APPROVED** [review]
Code is fine.

Corrrectly determines multiple instances on Windows.

**Comment by NickDraper** [comment]
Still needs testing via the package on linux and Mac

**Comment by keeeto** [comment]
I'm seeing a different error message on OSX now:
```
Traceback (most recent call last):
  File "psutil/__init__.py", line 149, in <module>
    import psutil._psosx as _psplatform
  File "psutil/_psosx.py", line 17, in <module>
    import _psutil_osx as cext
ImportError: No module named _psutil_osx
Unable to determine if other MantidPlot processes are running. Project recovery is disabled. Error msg: Python module psutil cannot be imported.
```

**Comment by OwenArnold** [comment]
I also get the same error as @keeeto 
```
Traceback (most recent call last):
  File "psutil/__init__.py", line 149, in <module>
    import psutil._psosx as _psplatform
  File "psutil/_psosx.py", line 17, in <module>
    import _psutil_osx as cext
ImportError: No module named _psutil_osx
Unable to determine if other MantidPlot processes are running. Project recovery is disabled. Error msg: Python module psutil cannot be imported.
```

**Comment by samueljackson92** [comment]
I also got the same error. This was because a couple of `*.so` files were not copied due to permissions. Adding a `chmod +w` step on those files seems to fit it for me.

**Comment by samueljackson92** [comment]
[This build](http://builds.mantidproject.org/job/pull_requests-osx/22150/) should have an artefact that can be used for testing.

**Comment by Pasarus** [comment]
I get the following error on boot up of Mantid on Ubuntu 16.04:
```Project Saving: Failed to get working folder whilst deleting checkpoints```

However I don't believe it is related to the other OSX issues.

Full Results log:
https://pastebin.com/NCPus6Ki

This is using this build and not the most recent:
http://builds.mantidproject.org/job/pull_requests-ubuntu/22700/

File: MantidPlot/make_package.rb.in [hunk 1]
```diff
@@ -326,6 +326,10 @@ end
 path = "/Library/Python/2.7/site-packages"
 directories = ["sphinx","sphinx_bootstrap_theme","IPython","zmq","pygments","backports", "qtawesome", "qtpy",
                "certifi","tornado","markupsafe","jinja2","psutil","jsonschema","functools32","ptyprocess","CifFile","yaml"]
+
+system("chmod +w #{path}/psutil/_psutil_osx.so")
```
Comment (martyngigg) [inline MantidPlot/make_package.rb.in]: I'm not sure I understand why not having write permissions on the source prevents copying?

**Review by keeeto — COMMENTED** [review]
This doesn't seem to solve the issue on my computer. I am still getting the same error.

**Review by Pasarus — CHANGES_REQUESTED** [review]
As stated previously it doesn't seem to work completely on Ubuntu, it does fix the overall issue but has introduced an unexpected output in the Debug Console ```Project Saving: Failed to get working folder whilst deleting checkpoints```

I've just checked and I have this on master aswell but it wasn't present on release branch until after this was fixed.

Edit: The error is being tripped in ProjectRecovery.cpp on line 110.

**Review by Pasarus — APPROVED** [review]

**Comment by martyngigg** [comment]
The proper fix for this is now in #23065.

**Pull request closed without merging**